### PR TITLE
bamboozling_beeble.txt and centaur_of_attention.txt and Support

### PR DESCRIPTION
**Pull request by Northmoc**
see #3943 for AI thoughts

### CODE REVIEW

**Comment by Northmoc** [comment]
Still need to clean up a few things on the centaur (AbilityUtils, prompt) but thought you might like to see the proof of concept

File: forge-gui/res/cardsfolder/b/barbarian_class.txt [hunk 1]
```diff
@@ -1,7 +1,9 @@
 Name:Barbarian Class
 ManaCost:R
 Types:Enchantment Class
-S:Mode$ Continuous | Affected$ You | AddKeyword$ If you would roll one or more dice, instead roll that many dice plus one and ignore the lowest roll. | Description$ If you would roll one or more dice, instead roll that many dice plus one and ignore the lowest roll.
+R:Event$ RollDice | ValidPlayer$ You | ReplaceWith$ PlusRoll | Description$ If you would roll one or more dice, instead roll that many dice plus one and ignore the lowest roll.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/b/barbarian_class.txt:4]: missing ActiveZones$

Comment (Northmoc) [reply forge-gui/res/cardsfolder/b/barbarian_class.txt:4]: I forgot replacement effects don't have the same zone defaults as statics

File: forge-gui/res/cardsfolder/b/bamboozling_beeble.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@ Types:Creature Beeble
 PT:1/1
 K:Protection from Robots
 A:AB$ Effect | Cost$ 1 T | ValidTgts$ Player | ReplacementEffects$ RigRoll | RememberObjects$ Targeted | ExileOnMoved$ Battlefield | SpellDescription$ The next time target player would roll one or more dice this turn, instead they roll that many dice plus one and you choose one of those rolls to ignore.
-SVar:RigRoll:Event$ RollDice | ValidPlayer$ Player.IsRemembered | ReplaceWith$ PlusRoll | Description$ The next time target player would roll one or more dice this turn, instead they roll that many dice plus one and you choose one of those rolls to ignore.
+SVar:RigRoll:Event$ RollDice | ActiveZones$ Command | ValidPlayer$ Player.IsRemembered | ReplaceWith$ PlusRoll | Description$ The next time target player would roll one or more dice this turn, instead they roll that many dice plus one and you choose one of those rolls to ignore.
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/b/bamboozling_beeble.txt:7]: ActiveZone Command is already set by EffectEffect:

https://github.com/Card-Forge/forge/blob/master/forge-game/src/main/java/forge/game/ability/effects/EffectEffect.java#L218

File: forge-game/src/main/java/forge/game/ability/AbilityUtils.java [hunk 3]
```diff
@@ -2625,6 +2625,28 @@ public static int xCount(final Card c, final String s, final CardTraitBase ctb)
             return doXMath(game.getStack().getMaxDistinctSources(), expr, c, ctb);
         }
 
+        if (sq[0].equals("MaxSameStoredRolls")) {
+            int max = 0;
+            List<Integer> rolls = c.getStoredRolls();
+            if (rolls != null) {
+                int lastNum = 1;
+                int tally = 0;
+                for (int i = 0; i < rolls.size(); i++) {
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ability/AbilityUtils.java:2634]: seems like basic for-each would be enough here
also I think it counts wrong in some edge cases (like all rolls being 1)

File: forge-game/src/main/java/forge/game/card/Card.java [hunk 4]
```diff
@@ -1279,6 +1280,32 @@ public final void moveMergedToSubgame(SpellAbility cause) {
...
+        return storedRolls;
+    }
+    public final List<String> getStoredRollsForView() {
+        List<String> forView = new ArrayList<>();
+        for (Integer i : storedRolls) {
+            forView.add(String.valueOf(i));
+        }
+        return forView;
+    }
+    public final void setStoredRolls(final List<Integer> results) {
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/card/Card.java:1293]: > If Centaur of Attention’s enters-the-battlefield ability is copied or triggers multiple times somehow, it’s possible to store more than five results on it. This is awesome. The last ability will consider all stored results and look for any that match each other.

File: forge-game/src/main/java/forge/game/ability/AbilityUtils.java (hunk 3 above)
Comment (tool4ever) [reply forge-game/src/main/java/forge/game/ability/AbilityUtils.java:2633]: `for (Integer roll : rolls)`

**Review by tool4ever — APPROVED** [review]
LGTM

**Pull request merged**